tests/unit/file/read-file-spec.js

```diff
@@ -0,0 +1,86 @@
+var rewire = require('rewire');
+var read = rewire('../../../src/file/read-file');
+
+describe('read file', function() {
+
+    var text,
+        extractedCode;
+
+    beforeEach(function() {
+
+        spyOnReadTextFile();
+        spyOnSourceMapExtract();
+
+    });
+
+    var spyOnReadTextFile = function() {
+
+        read.__set__('readTextFile', function(filePath, cb) {

```

Comment:
Here I mock out the `readTextFile` function so we can write this as a unit test instead of creating a file on disk first. This code not mocked out is flexed by the integration tests.